app/policies/proposal_policy.rb

```diff
@@ -13,24 +14,60 @@ def initialize(user, proposal)
     @proposal = proposal
   end
 
-  def author?
+  def is_author?
     @proposal.requester_id == @user.id
   end
 
-  def not_approved?
+  def is_not_approved?
     !@proposal.approved?
   end
 
-  def approve_reject?
+  def is_approver?
+    !@proposal.approvals.find_by(user: @user).nil?
+  end
+
+  def is_observer?
+    @proposal.observers.include? @user
+  end
+
+  def is_pending_approver?
     actionable_approvers = @proposal.currently_awaiting_approvers
     actionable_approvers.include? @user
   end
 
-  def edit?
-    self.test_all(:edit?)
+  def can_approve_or_reject?
+    self.test_all(:can_approve_or_reject?)
+  end
+
+  def can_edit?
+    self.test_all(:can_edit?)
   end
 
-  def update?
-    self.edit?
+  def can_update?
+    self.can_edit?
+  end
+
+  def can_show?
+    self.is_author? || self.is_approver? || self.is_observer?
+  end
+
+  # equivalent of can_show?
+  class Scope
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+
+    def resolve
+      # Use subselects instead of left joins to avoid duplication removal
```

Comment:
A better comment may be 
```ruby
# use subselects instead of left joins to avoid an explicit duplication-removal step
```

If we were to use an inner join, we'd be skipping over carts with no observers or no approvers (as they'd be filtered out). If we instead used a left join, we'd get one row for every cart x cart's observer x cart's approver tuple. The duplicates can be removed with a `distinct`, but that makes the query even more complicated, and may make it play less well with further filters.

Instead, this uses subselects, which require explicit sql, but should make working with the results much easier. Postgres should also be smart enough to make these subselects efficient.